packages/shared/sierra-client/src/email-verification-notes.ts

```diff
@@ -0,0 +1,131 @@
+import { getVarFieldContent, VarField, varFieldTags } from './patron';
+
+const auth0NotePrefix = 'Auth0:';
+
+// See https://stackoverflow.com/a/9204568
+// Matches string@string.string
+// There are some email addresses in Sierra that don't match this - because of whitespace,
+// nonsense values, NONE values, etc. We're OK with not matching them because we need to be
+// able to look them up and deal with them as email addresses, regardless.
+const emailRegex = '\\S+@\\S+\\.\\S+';
+
+// See https://stackoverflow.com/a/3143231
+// Matches ISO 8601 dates like YYYY-MM-DDTHH:MM:SS.SSSSZ
+const isoDateRegex =
+  '\\d{4}-[01]\\d-[0-3]\\dT[0-2]\\d:[0-5]\\d:[0-5]\\d\\.\\d+(?:[+-][0-2]\\d:[0-5]\\d|Z)';
+
+// The distinction here is that we assume that users who exist already in Sierra
+// are verified, because they must've used their registered email address to set a password.
+// We distinguish between them in the message to maximise readability and to ease debugging.
+const messages = {
+  verified: 'verified at',
+  assumedVerified: 'assumed to be verified at login',
```

Comment:
Sorry to bikeshed, but looking through subsequent PRs I'm falling out of love with this "assumed" naming. The further away you get, the less obvious it is what this means. Let's make it super obvious so these notes make sense even when this code is long and gone.

Maybe we could call it explicit/implicit verification? Then you get:

```typescript
const messages = {
  explicitVerification: 'explicitly verified by the user at',
  implicitVerification: 'implicitly verified on initial Auth0 login with an existing Sierra account at',
}
```

Reply:
Yeah, I think I discounted this naming at some point but actually it looks fine now - will change that